### ISSUE TITLE
Listchar nbsp should also apply to other unicode spaces (e.g. Em Space U+2003)

### ISSUE THREAD

**Issue by Cynosure-North** [vim/vim#12014]
There are a number of invisible unicode characters:

    Em Space: U+2003
    Tag Space: U+E0020
    Thin Space: U+2009
    En Space: U+2002
    Hair Space: U+200A
    Zero Width No-Break Space: U+FEFF
    Figure Space: U+2007
    Punctuation Space: U+2008
    Six-Per-Em Space: U+2006
    Four-Per-Em Space: U+2005
    Three-Per-Em Space: U+2004
    Ideographic Space: U+3000
    Medium Mathematical Space: U+205F

These characters are invisible, but not necessarily considered whitespace, which can lead to numerous issues.
Vim already has a mechanism to handle this (listchars), but currently there is no option to highlight these characters.

**Expected Behaviour**
The character set for nbsp should also be displayed for these other invisible characters.

**Alternatives**
I can't think of any better alternatives.

**Comment by ychin** [comment]
I don't think those spaces are actually non-breaking spaces (nbsp) though? Seems like we wouldn't want to hijack the meaning of that word, and the issue here is there are a lot of different kinds of spaces in Unicode, but I'm not sure if we want them to all be replaced with the same listchars character. This is particularly important because sometimes the reason I would use `listchars` to begin with is to see if someone injected a wrong space character in my document to begin with. If all kinds of spaces show up as the same listchar character I think that decreases its utility a little bit.

**Comment by chrisbra** [comment]
Yes, I think this was discussed before and the general agreement was to use `nbsp` only for a realy non-breaking space and none of the other UTF-8 special whitespace characters.

**Comment by brammool** [comment]
> There are a number of invisible unicode characters:
> 
>     Em Space: U+2003
>     Tag Space: U+E0020
>     Thin Space: U+2009
>     En Space: U+2002
>     Hair Space: U+200A
>     Zero Width No-Break Space: U+FEFF
>     Figure Space: U+2007
>     Punctuation Space: U+2008
>     Six-Per-Em Space: U+2006
>     Four-Per-Em Space: U+2005
>     Three-Per-Em Space: U+2004
>     Ideographic Space: U+3000
>     Medium Mathematical Space: U+205F
> 
> These characters are invisible, but not necessarily considered
> whitespace, which can lead to numerous issues.

Some of these actually show, but are not a full cell width.

> Vim already has a mechanism to handle this (listchars), but currently
> there is no option to highlight these characters.
> 
> **Expected Behaviour**
> The character set for nbsp should also be displayed for these other
> invisible characters.

The "nbsp" is used quite often in HTML.  Most of the characters you list
are unusual.  I think they deserve a different entry.  If the user
doesn't care the same replacement character can be used, but if the user
does want to see the difference we do need another entry.

Perhaps we can group these into a small number of categories?


-- 
hundred-and-one symptoms of being an internet addict:
156. You forget your friend's name but not her e-mail address.

 /// Bram Moolenaar -- ***@***.*** -- http://www.Moolenaar.net   \\\
///                                                                      \\\
\\\        sponsor Vim, vote for features -- http://www.Vim.org/sponsor/ ///
 \\\            help me help AIDS victims -- http://ICCF-Holland.org    ///
